src/Spectre.libGenetic.Tests/GenerationFactoryTest.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+* GenerationFactoryTest.cpp
+* Tests GenerationFactory.
+*
+Copyright 2017 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <gtest/gtest.h>
+#include "Spectre.libGenetic/GenerationFactory.h"
+#include "Spectre.libException/ArgumentOutOfRangeException.h"
+
+namespace
+{
+using namespace spectre::algorithm::genetic;
+using namespace spectre::core::exception;
+
+TEST(GenerationFactoryInitialization, initializes)
+{
+    const auto generationSize = 5u;
+    const auto individualSize = 6u;
+    const auto initialFillup = 4u;
+    EXPECT_NO_THROW(GenerationFactory(generationSize, individualSize, initialFillup));
+}
+
+TEST(GenerationInitialization, throws_when_required_number_of_active_is_bigger_than_length)
+{
+    const auto generationSize = 5u;
+    const auto individualSize = 4u;
+    const auto excessiveInitialFillup = 6u;
+    EXPECT_THROW(GenerationFactory(generationSize, individualSize, excessiveInitialFillup), ArgumentOutOfRangeException<size_t>);
+}
+
+class GenerationFactoryInitializationTest : public ::testing::Test
+{
+public:
+    GenerationFactoryInitializationTest():
+        generationFatory(generationSize, individualSize, initialFillup),
+        generationFatoryWithBigIndividuals(bigGenerationSize, bigIndividualSize, bigInitialFillup) {}
+
+protected:
+    const size_t generationSize = 1u;
+    const size_t individualSize = 6u;
+    const size_t initialFillup = 4u;
+    const size_t bigGenerationSize = 2u;
+    const size_t bigIndividualSize = 100u;
+    const size_t bigInitialFillup = 60u;
+    const Seed seed = 0ul;
```

Comment:
unsigned long? :smiley:

Reply:
Again, I dont't understand :D. Types are coherent with ones that are parameters of the functions.

Reply:
@Sand3r-, this one was `unsigned long` for current RNG type indeed.

Reply:
nothing to do here, so done

Reply:
@gmrukwa Funny how other seeds are just unsigned ints then xd.